include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -826,21 +857,26 @@ template <typename _BackendTag, typename _ExecutionPolicy, typename _Range, type
 oneapi::dpl::__internal::__difference_t<_Range>
 __pattern_unique(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _Range&& __rng, _BinaryPredicate __pred)
 {
-    if (__rng.size() == 0)
-        return __rng.size();
+    const auto __n = oneapi::dpl::__ranges::__size(__rng);
+    if (__n == 0)
+        return __n;
 
     using _ValueType = oneapi::dpl::__internal::__value_t<_Range>;
 
-    oneapi::dpl::__par_backend_hetero::__buffer<_ValueType> __buf(__rng.size());
-    auto res_rng = oneapi::dpl::__ranges::views::all(__buf.get_buffer());
+    auto&& __rng_s = oneapi::dpl::__ranges::__get_subscription_view(__rng);
```

Comment:
Is not it better to use `oneapi::dpl::__ranges::__get_subscription_view __rng_s(std::forward<_Range>(__rng));`?
I do not quite understand the purpose of `auto&&` in the context of calling a constructor.